storage.go

```diff
@@ -4,37 +4,39 @@ import (
 	"context"
 )
 
-// Storage represents the abstraction for underlying storage backends. A storage
-// implementation does not care about specific data types. All the
-// storage cares about are key-value pairs. Code making use of the storage
-// have to take care about specific types they care about them self.
+// KV is a key-value pair.
+type KV struct {
+	// Key should take a format of path separated by slashes "/". E.g.
+	// "/a/b/c". Storage implementations should validate the key before
+	// storing them. Key sanitized before inserting to the storage. I.e.
+	// leading slash can be added and trailing slash can be removed. E.g.
+	// "/a/b/c", "a/b/c/", "a/b/c/", and "/a/b/c/" represent the same key.
+	Key string
+	// Val is an arbitrary value stored under the Key.
+	Val string
+}
+
+func NewKV(key, val string) KV {
```

Comment:
@JosephSalisbury I tired that, it doesn't work for methods like `Search` or `Exists` where you still have to validate the key. So I leave it as it is, unless you have some nice idea.